kartothek/core/_deprecation.py

```diff
@@ -149,3 +149,20 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return _deprecate_kwarg
+
+
+def deprecate_multi_table_read(func, *, stacklevel=2):
```

Comment:
While I can see the technical elegance of our function wrappers, it makes our code really badly readable / understandable.

Reply:
Generally, I would agree with you. But we are not changing any actual functionality here, so rather than calling a function which emits the warning or copy-pasting code inside of each `read_dataframes_*` function, I think this is more appropriate as we are keeping the actual function code limited to the behavior of the function instead of clobbering it with deprecation code.